### PR TITLE
Consent Mechanism Update

### LINKED ISSUE #3041: DB: Change ConsentMechanism "Necessary" to "Notice Only"
### Is your feature request related to a specific problem?

Consent Mechanism "Necessary" is changing to "Notice Only"

_A description of what you want to happen._

Add a migration that does the following:

1) Add “notice_only” enum to list 
2) Update all privacy notice and privacy notice history records that have a consent_mechanism of “necessary” to be “notice_only”.  3) Remove “necessary” as an option.

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by pattisdr**
Closes #3041 

### Code Changes

* [ ] Schema/Data migration that updates the existing consentmechanism enum and any records in PrivacyNotice/PrivacyNoticeHistory that previously used the "necessary" value to be notice only
* [ ] While I'm here, fix the previous migration for users that already had privacy notices saved (unlikely) to better handle adding and removing nullable fields

### Steps to Confirm

* [ ] Run `nox -s dev -- shell`
* [ ] Get some privacy notices in your db **before checking out this branch**.  Use the postman collection to add Privacy Notices with "necessary" consent mechanism
* [ ] Checkout this branch
* [ ] cd src/fides/api/ctl
* [ ] Verify in db privacy notices and privacy notice history have `notice_only` consent mechanism now (if automigrated)
* [ ] `alembic downgrade -1`
* [ ] Verify in db privacy notices and privacy notice history have "necessary" consent mechanism
* [ ] `alembic upgrade head`
* [ ] Verify in db privacy notices and privacy notice history have `notice_only` consent mechanism now again
* [ ] `select enum_range(NULL::consentmechanism);` should have opt in/opt out/notice only
* [ ] `alembic downgrade -2` with privacy notices and privacy notice histories should work without issue
* [ ] On `alembic upgrade head` verify that new `displayed_in_overlay` and `displayed_in_api` fields don't have server defaults

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

Updates ConsentMechanism Enum to be **"opt_in", "opt_out", and "notice_only"** instead of "opt_in", "opt_out", and "necessary".

### CODE REVIEW

**Comment by pattisdr** [comment]
@adamsachs tagged you for review since I'm editing your model and @allisonking for awareness since https://ethyca.atlassian.net/wiki/spaces/EN/pages/2649096193/2023-03-21+Frontend+Design%3A+Privacy+Notices?focusedCommentId=2665185284 is the conversation that prompted this change

File: src/fides/api/ctl/migrations/versions/144d9b85c712_update_consent_mechanism.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+
+def upgrade():
+    """
+    Extra steps added here to avoid the error that enums have to be committed before they can be used.  This
+    avoids having to commit in the middle of this migration and lets the entire thing be completed in a single transaction
+    """
+    op.execute("alter type consentmechanism rename to consentmechanismold")
+
+    op.execute(
```
Comment (adamsachs) [inline src/fides/api/ctl/migrations/versions/144d9b85c712_update_consent_mechanism.py:24]: oh, this is clever! what i've [done in the past](https://github.com/ethyca/fides/blob/main/src/fides/api/ctl/migrations/versions/2f6aa5fe797a_update_messaging_service_type_enum.py#L29) is create a temporary column, which is similar but feels a bit clunkier than what you have here

**Review by adamsachs — APPROVED** [review]
looks good! clever migration tactic @pattisdr

File: src/fides/api/ctl/migrations/versions/144d9b85c712_update_consent_mechanism.py (hunk 1 above)
Comment (pattisdr) [reply src/fides/api/ctl/migrations/versions/144d9b85c712_update_consent_mechanism.py:24]: awesome, thanks for your review!

**Pull request merged**